environment.yml

```diff
@@ -4,11 +4,14 @@ channels:
     - bioconda
     - defaults
 dependencies:
-    - fastp
-    - fastqc
-    - gzip
-    - mash
-    - quast
-    - seqtk
-    - snakemake
-    - spades
+    - black>=21.10b0
+    - fastp>=0.23.1
+    - fastqc>=0.11.9
+    - gzip>=1.7
+    - mash>=2.3
+    - pytest-cov>=3.0.0
+    - quast>=5.0.2
+    - seqtk>=1.3
+    - snakemake>=6.10.0
+    - spades>=3.15.3
+    - versioneer>=0.21
```

Comment:
This looks mostly reasonable, but a few comments about pinning version numbers.
- As a rule, you'll probably want to pin major.minor version numbers (X.Y) and not major.minor.patch numbers (X.Y.Z). By definition, patch updates are typically going to be minor bug fixes. There may be exceptions to this for major version number 0 (a.k.a. prior to a 1.0 release) when there's a bit more flexibility with following versioning semantics.
- You've done the right thing by setting *minimum* version numbers for these packages instead of *exact* version numbers. But the Black library is an exception: typically you'll need to pin that to an exact version, since subsequent updates to Black can cause an unexpected `make style` failure.
- spades 3.15.3 is not available for Mac on Conda. I fixed this by changing the minimum version to 3.15.2, but this would be solved by just specifying 3.15 as mentioned above.

Reply:
Good to know about not including patch numbers!
You are absolutely right about blacker; it makes sense.